### PR TITLE
Fix VPP coverity issues

### PR DESCRIPTION
**Pull request by Alex1Zhang**
Signed-off-by: Alex1 Zhang alex1.zhang@intel.com

### CODE REVIEW

File: videoprocess/vpphdr_tm.cpp [hunk 1]
```diff
@@ -588,7 +590,9 @@ bool read_frame_to_surface(FILE *fp, VASurfaceID surface_id)
         frame_size = va_image.width * va_image.height * 4;
         src_buffer = (unsigned char*)malloc(frame_size);
         assert(src_buffer);
-        fread(src_buffer, 1, frame_size, fp);
+        do {//
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:593]: deadloop if the file is corrupted?

File: videoprocess/vpphdr_tm.cpp [hunk 2]
```diff
@@ -560,7 +560,9 @@ bool read_frame_to_surface(FILE *fp, VASurfaceID surface_id)
 
         src_buffer = (unsigned char*)malloc(frame_size);
         assert(src_buffer);
-        fread(src_buffer, 1, frame_size, fp);
+        do{
+            n_items = fread(src_buffer, 1, frame_size, fp);
```
Comment (XinfengZhang) [inline videoprocess/vpphdr_tm.cpp:564]: what will happen if the src file is corrupted. and full size < frame_size?

Comment (Alex1Zhang) [reply videoprocess/vpphdr_tm.cpp]: add log to check the return value instead of read in loop.

**Review by FurongZhang — APPROVED** [review]
Looks good to me.

File: videoprocess/vpp3dlut.cpp [hunk 3]
```diff
@@ -1236,10 +1236,6 @@ video_frame_process_scaling(VASurfaceID in_surface_id,
         vaDestroyBuffer(va_dpy, pipeline_param_buf_id);
     }
 
-    if (filter_param_buf_id != VA_INVALID_ID) {
```
Comment (XinfengZhang) [inline videoprocess/vpp3dlut.cpp:1239]: where to destroy filter_param_buf_id?

Comment (Alex1Zhang) [reply videoprocess/vpp3dlut.cpp:1239]: triage the deadcode error, this code always not be entered, so remove it.

**Review by XinfengZhang — APPROVED** [review]

**Pull request merged**